internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+	return msgExprLiteral
+}
+
+// ConvertCommandToMessageConstantBooleanExpr converts a command.Expr to a message.Expr_Constant
+func ConvertCommandToMessageConstantBooleanExpr(cmd *command.ConstantBooleanExpr) *Expr_Constant {
+	msgExprConstant := &Expr_Constant{&ConstantBooleanExpr{}}
+	msgExprConstant.Constant.Value = cmd.Value
+	return msgExprConstant
+}
+
+// ConvertCommandToMessageUnaryExpr converts a command.Expr to a message.Expr_Unary
+func ConvertCommandToMessageUnaryExpr(cmd *command.UnaryExpr) *Expr_Unary {
+	msgExprUnary := &Expr_Unary{&UnaryExpr{}}
+	msgExprUnary.Unary.Operator = cmd.Operator
+	msgExprUnary.Unary.Value = ConvertCommandToMessageExpr(cmd.Value)
+	return msgExprUnary
+}
+
+// ConvertCommandToMessageBinaryExpr converts a command.Expr to a message.Expr_Binary
+func ConvertCommandToMessageBinaryExpr(cmd *command.BinaryExpr) *Expr_Binary {
+	msgExprBinary := &Expr_Binary{&BinaryExpr{}}
+	msgExprBinary.Binary.Operator = cmd.Operator
+	msgExprBinary.Binary.Left = ConvertCommandToMessageExpr(cmd.Left)
+	msgExprBinary.Binary.Right = ConvertCommandToMessageExpr(cmd.Right)
+	return msgExprBinary
+}
+
+// ConvertCommandToMessageRepeatedExpr converts a []command.Expr to a message.Expr
+func ConvertCommandToMessageRepeatedExpr(cmd []command.Expr) []*Expr {
+	msgRepeatedExpr := []*Expr{}
+	for i := range cmd {
+		msgRepeatedExpr = append(msgRepeatedExpr, ConvertCommandToMessageExpr(cmd[i]))
+	}
+	return msgRepeatedExpr
+}
+
+// ConvertCommandToMessageFunctionalExpr converts a command.Expr to a message.Expr_CFExpr_Func
+func ConvertCommandToMessageFunctionalExpr(cmd *command.FunctionExpr) *Expr_Func {
+	msgExprFunc := &Expr_Func{&FunctionExpr{}}
+	msgExprFunc.Func.Name = cmd.Name
+	msgExprFunc.Func.Distinct = cmd.Distinct
+	msgExprFunc.Func.Args = ConvertCommandToMessageRepeatedExpr(cmd.Args)
+	return msgExprFunc
+}
+
+// ConvertCommandToMessageEqualityExpr converts a command.Expr to a message.Expr_Equality
+func ConvertCommandToMessageEqualityExpr(cmd *command.EqualityExpr) *Expr_Equality {
+	msgExprEquality := &Expr_Equality{&EqualityExpr{}}
+	msgExprEquality.Equality.Left = ConvertCommandToMessageExpr(cmd.Left)
+	msgExprEquality.Equality.Right = ConvertCommandToMessageExpr(cmd.Right)
+	msgExprEquality.Equality.Invert = cmd.Invert
+	return msgExprEquality
+}
+
+// ConvertCommandToMessageRangeExpr converts a command.Expr to a message.Expr_Range
+func ConvertCommandToMessageRangeExpr(cmd *command.RangeExpr) *Expr_Range {
+	msgExprRange := &Expr_Range{&RangeExpr{}}
+	msgExprRange.Range.Needle = ConvertCommandToMessageExpr(cmd.Needle)
+	msgExprRange.Range.Lo = ConvertCommandToMessageExpr(cmd.Lo)
+	msgExprRange.Range.Hi = ConvertCommandToMessageExpr(cmd.Hi)
+	msgExprRange.Range.Invert = cmd.Invert
+	return msgExprRange
+}
+
+// ConvertCommandToMessageExpr converts command.Expr to a message.Expr
+func ConvertCommandToMessageExpr(cmd command.Expr) *Expr {
+	msgExpr := &Expr{}
+	switch cmd := cmd.(type) {
```

Comment:
use another variable for that, such as `c` instead of a second `cmd`

Reply:
golint suggested its use actually.